File: Backtrack/subsets.js
/**78
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function (nums) {
  let result = [];

  let backtrack = (path, start) => {
    result.push([...path]);
    for (let i = start; i < nums.length; i++) {
      path.push(nums[i]);
      backtrack(path, i + 1);
      path.pop();
    }
  };
  backtrack([], 0);
  return result;
};

File: Backtrack/subsets.cs
public class Solution1 {
    public IList<IList<int>> Subsets(int[] nums)
    {
        IList<IList<int>> list = [];

        void backtrack(List<int> path, int start)
        {
            list.Add([.. path]);
            for (var i = start; i < nums.Length; i++)
            {
                path.Add(nums[i]);
                backtrack(path, i + 1);
                path.RemoveAt(path.Count - 1);
            }
        }

        backtrack([], 0);
        return list;
    }
}